includes/permissions.sh

```diff
@@ -0,0 +1,31 @@
+# shellcheck disable=SC2148
+
+# Changes the ownership to $APP_USER:$APP_GROUP for the given file / recursively to the given folder
+change_ownership() {
+    local APP_USER="$1"
+    local APP_GROUP="$2"
+    local TARGET="$3"
+
+    # Find files or directories not owned by specified user and group
+    files_with_incorrect_permissions=$(find "$TARGET" ! -user "$APP_USER" -o ! -group "$APP_GROUP")
+
+    if [ -z "$files_with_incorrect_permissions" ]; then
+        ei "> All items in $TARGET are already owned by $APP_USER:$APP_GROUP."
+    else
+        # Echo the total count of files_with_incorrect_permissions
+        count=$(echo "$files_with_incorrect_permissions" | wc -l)
+        ei "> Found $count items with improper permissions for $TARGET"
+
+        # Check if running as root and warn user if not
+        if [ "$EUID" -ne 0 ]; then
```

Comment:
What is an EUID?

Reply:
EUID is the effective user id so I thought it best to use that to check that to detect root.

This post explains my current understanding https://www.reddit.com/r/linux4noobs/comments/amg0fg/a_question_about_process_uids_and_euids/

Reply:
Try this https://stackoverflow.com/a/27670422 and and we use gosu not sudo, dont think the EUID will catch for that. Even if it would, the PUID would be the same ID as the other one.